sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -186,6 +190,13 @@ class CodegenContext {
       freshNameIds += fullName -> 1
       fullName
     }
+    freshNameJavaTypes.put(freshname, javaType)
+    freshname
+  }
+
+  def isColumnarType(name: String): Boolean = {
```

Comment:
The bottom line is that we need to maintain this for each variable. For example, ```scanBatches()```, we need to access ```InternalRow``` for a column with ```struct``` type (e.g. ```ColumnVector.getStruct()``` is called. In my first prototype, I set ```true``` into ```ctx.columnarInput```. However, I cannot support such a complex case.

I will replace lookup this by name with lookup this by setting a flag in `freshName()` to avoid brittleness.